src/transformers/modeling_utils.py

```diff
@@ -619,6 +621,11 @@ def _load_state_dict_into_meta_model(
                 and dtype == torch.float16
             ):
                 param = param.to(torch.float32)
+                force_upcast_dtype = torch.float32
+
+                # For backward compatibility with older versions of `accelerate`
+                if set_module_tensor_to_device.__code__.co_argcount == 5:
```

Comment:
Slight nit: can we use the signature and parameter names using `inspect`? It would be clearer to read. Also add a TODO that this should become a version check at the next version of Accelerate (I will take care of it after next release).

Reply:
Thanks! Should be addressed in 95486c3